sepal_ui/mapping/sepal_map.py

```diff
@@ -778,28 +683,132 @@ def get_viz_params(image):
+        If base is set to True, the basemaps are removed as well
+
+        Args:
+            base (bool, optional): wether or not the basemaps should be removed, default to False
+        """
+        # filter out the basemaps if base == False
+        layers = self.layers if base else [lyr for lyr in self.layers if not lyr.base]
+
+        # remove them using the layer objects as keys
+        [self.remove_layer(layer, base) for layer in layers]
+
+        return
+
     def add_layer(self, layer, hover=False):
-        """Add layer and use a default style for the GeoJSON inputs
+        """
+        Add layer and use a default style for the GeoJSON inputs.
+        Remove existing layer if already on the map.
 
+        layer (ipyleaflet.Layer): any layer type from ipyleaflet
         hover (bool): whether to use the default hover style or not.
+        """
+
+        # remove existing layer before addition
+        existing_layer = self.find_layer(layer.name)
+        not existing_layer or self.remove_layer(existing_layer)
+
+        # apply default coloring for geoJson
+        if isinstance(layer, ipl.GeoJSON):
+            layer.style = layer.style or styles.layer_style
+            hover_style = styles.layer_hover_style if hover else layer.hover_style
+            layer.hover_style = layer.hover_style or hover_style
+
+        super().add_layer(layer)
+
+        return
 
+    def add_basemap(self, basemap="HYBRID"):
         """
+        Adds a basemap to the map.
 
-        if isinstance(layer, GeoJSON):
+        Args:
+            basemap (str, optional): Can be one of string from basemaps. Defaults to 'HYBRID'.
+        """
+        if basemap not in basemap_tiles.keys():
+            keys = "\n".join(basemap_tiles.keys())
+            msg = f"Basemap can only be one of the following:\n{keys}"
+            raise ValueError(msg)
 
-            if not layer.style:
-                layer.style = styles.layer_style
+        self.add_layer(basemap_tiles[basemap])
 
-            if hover and not layer.hover_style:
-                layer.hover_style = styles.layer_hover_style
+        return
 
-        super().add_layer(layer)
+    def get_scale(self):
+        """
+        Returns the approximate pixel scale of the current map view, in meters.
+        Reference: https://blogs.bing.com/maps/2006/02/25/map-control-zoom-levels-gt-resolution
 
-        return self
+        Returns:
+            (float): Map resolution in meters.
+        """
+
+        return 156543.04 * math.cos(0) / math.pow(2, self.zoom)
+
+    def find_layer(self, key, base=False):
+        """
+        Search a layer by name or index
+
+        Args:
+            key (Layer, str, int): the layer name, index or directly the layer
+            base (bool, optional): either the basemaps should be included in the search or not. default t false
+
+        Return:
+            (TileLLayerayer): the first layer using the same name or index else None
```

Comment:
Returning `None` instead of an exception could hide some errors, perhaps do you have a use case where you might be interested on that return?

Reply:
I don't have any in mind but what if we do as in `pathlib.Path.mkdir`: 
- default behaviour: raise an error when the result is None
- if the parameter `none_ok` is True go on with nothing